src/vm/eventpipe.cpp

```diff
@@ -732,6 +711,50 @@ StackWalkAction EventPipe::StackWalkCallback(CrawlFrame *pCf, StackContents *pDa
     return SWA_CONTINUE;
 }
 
+uint32_t EventPipe::GenerateSessionIndex()
+{
+    LIMITED_METHOD_CONTRACT;
+    PRECONDITION(IsLockOwnedByCurrentThread());
+
+    uint64_t id = 1;
+    for (uint32_t i = 0; i < MaxNumberOfSessions; ++i, id <<= 1)
+    {
+        if ((s_activeSessions & id) == 0)
+        {
+            s_activeSessions |= id;
+            return i;
+        }
+    }
+    return MaxNumberOfSessions;
+}
+
+void EventPipe::ResetSessionIndex(uint64_t mask)
+{
+    LIMITED_METHOD_CONTRACT;
+    PRECONDITION((s_activeSessions & mask) != 0);
```

Comment:
s_activeSessions & mask == mask?

Reply:
The comment should go away once we replace `s_activeSessions` by `s_sessionCount`.